src/jdk.attach/linux/classes/sun/tools/attach/VirtualMachineImpl.java

```diff
@@ -378,6 +320,72 @@ private long getNamespacePid(long pid) throws AttachNotSupportedException, IOExc
         }
     }
 
+    private static final String FIELD = "field";
+    private static final String MASK  = "mask";
+
+    private static final String SIGNAL_MASK_PATTERN = "(?<" + FIELD + ">Sig\\p{Alpha}{3}):\\s+(?<" + MASK + ">\\p{XDigit}{16}).*";
+
+    private static final long SIGQUIT = 1L << 2;
+
+    private static boolean checkCatchesAndSendQuitTo(int pid, boolean throwIfNotReady) throws AttachNotSupportedException, IOException {
```

Comment:
Looks like the throwIfNotReady param is not needed, can be removed.

Reply:
I changed the logic so that now when the timeout loop times out the last attempt to send the QUIT will throw if the target is not catching QUIT